geofence/geofence.go

```diff
@@ -0,0 +1,66 @@
+package echogeofence
+
+import (
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/circa10a/go-geofence"
+	"github.com/labstack/echo/v4"
+)
+
+// EchoGeofence holds a geofence config
+type EchoGeofence geofence.Config
+
+// config returns a geofence config from go-geofence
+func (e *EchoGeofence) config() *geofence.Config {
+	return &geofence.Config{
+		IPAddress: e.IPAddress,
+		Token:     e.Token,
+		Radius:    e.Radius,
+		CacheTTL:  e.CacheTTL,
+	}
+}
+
+// Middleware looks up the coordinates of a client to see if it's nearby
+func Middleware(echoGeofence *EchoGeofence) echo.MiddlewareFunc {
```

Comment:
Maybe instead of passing `*EchoGeofence` we should pass interface that object created by `geofence.New(echoGeofence.config())` satisfies. This way we can delegate all configuration related stuff to geofence library and do not need to update this middleware if something changes with `go-geofence`

```go
type IsAddressNearChecker interface {
	IsIPAddressNear(ipAddress string) (bool, error)
}

func Middleware(geofence IsAddressNearChecker) echo.MiddlewareFunc {
```

Reply:
Makes sense. Made the changes